corehq/apps/receiverwrapper/views.py

```diff
@@ -205,6 +206,11 @@ def _record_metrics(tags, submission_type, response, timer=None, xform=None):
     metrics_counter('commcare.xform_submissions.count', tags=tags)
 
 
+@api_auth
```

Comment:
I think this also needs these other decorators: `waf_allow`, `csrf_exempt`, `location_safe` (I think it should actually be `location_safe_bypass`):

```suggestion
@waf_allow('XSS_BODY')
@csrf_exempt
@location_safe_bypass
@api_auth
```

Reply:
Why `@location_safe_bypass`?  I'd think it'd be appropriate to say this endpoint is only available to users who don't have their access restricted by location.

Reply:
That's fair. The current endpoint is marked with `location_safe` which should actually be `location_safe_bypass` since it does not perform any location checks. But you're right that this endpoint should not have the `location_safe` decorator at all. I'll update the comment.